tests/codeception/acceptance/Content/NewsCest.php

```diff
@@ -254,4 +255,94 @@ public function testMetaData(AcceptanceTester $I) {
     $I->assertEquals($values['featured_image_alt'], $I->grabAttributeFrom('meta[name="twitter:image:alt"]', 'content'), 'Metadata "twitter:image:alt" should match.');
   }
 
-}
+  #[CodeceptionAttribute\Group('news_variant')]
+  /**
+   * Test that news variants display correct fields and can be populated with site manaager role.
+   */
+  public function testNewsVariantFieldsDisplay(AcceptanceTester $I) {
+    // Test with site-manager role
+    $I->logInWithRole('site_manager');
+    $I->amOnPage('/node/add/stanford_news');
+    $I->canSeeResponseCodeIs(200);
+    
+    // Generate test data using faker
+    $testData = [
+      'Dek' => $this->faker->sentence(),
+      'Byline' => $this->faker->name(),
+      'Banner Caption' => $this->faker->sentence(),
+      'Quote' => $this->faker->sentence(),
+      'Subtitle' => $this->faker->words(4, TRUE)
+    ];
+
+    // Fill in required title field
+    $I->fillField('Headline', $this->faker->words(3, TRUE));
+    
+    // Verify fields are initially empty using canSeeInField()
+    $I->canSeeInField('Dek', '');
+    $I->canSeeInField('Byline', '');
+    $I->canSeeInField('Banner Caption', '');
+    $I->canSeeInField('Quote / Big Text', '');
+    $I->canSeeInField('Subtitle', '');
+    
+    // Fill in the fields
+    $I->fillField('Dek', $testData['Dek']);
+    $I->fillField('Byline', $testData['Byline']);
+    $I->fillField('Banner Caption', $testData['Banner Caption']);
+    $I->fillField('Quote / Big Text', $testData['Quote']);
+    $I->fillField('Subtitle', $testData['Subtitle']);
+    
+    // Save the node
+    $I->click('Save');
+    $I->canSeeResponseCodeIs(200);
+    $I->canSee('has been created');
+    
+    // Verify the information is reflected on the page
+    $I->canSee($testData['Dek']);
+    $I->canSee($testData['Byline']);
+    $I->canSee($testData['Banner Caption']);
+    // $I->canSee($testData['Quote']);
+    // $I->canSee($testData['Subtitle']);
+  }
+
+  /**
+   * Test that news variants display correct fields and can be populated with contributor role.
+   */
+  public function testNewsVariantFieldsDisplayContributor(AcceptanceTester $I) {
```

Comment:
It looks to me like this is a redundant test.  If we need to run the same test with multiple roles, can we refactor it something like: 

```
foreach $role in $roles {
  $I->logInWithRole($role);
   // do test
}
```

Reply:
Great catch! The test has been refactored. Please give it another glance and let me know if any additional tweaks are needed. Thanks @imonroe! 😄